src/mom5/ocean_csiro_bgc/csiro_bgc.F90

```diff
@@ -1064,9 +1075,10 @@ subroutine csiro_bgc_sbc(isc, iec, jsc, jec, isd, ied, jsd, jed, &
 
 do j = jsc, jec  !{
   do i = isc, iec  !{
-    sc_co2(i,j) = 2073.1 + t_prog(indtemp)%field(i,j,1,time%taum1) * &
-         (-125.62 + t_prog(indtemp)%field(i,j,1,time%taum1) *        &
-          (3.6276 + t_prog(indtemp)%field(i,j,1,time%taum1) *        &
+    schmidt_temp = min(35.0, t_prog(indtemp)%field(i,j,1,time%taum1))! add max limit to temp used for Schmidt number calculation.  mac, jan21.
+    sc_co2(i,j) = 2073.1 + schmidt_temp * &
```

Comment:
will this give numerically identical results to the old `sc_co2` calculation?

Reply:
Good point, the Schmidt Number and gas exchange will not be identical wherever there is a surface temperature over 35 degC.

The change was made for the ESM after some points in extended warm climate CMIP6 experiments reached about 42 degC, where the Schmidt number went negative and blew up the model, as found by Tilo Z. and Martin D. in late 2020.
Checking the reference to the Schmidt formula (Wanninkhof 1992, https://agupubs.onlinelibrary.wiley.com/doi/epdf/10.1029/92JC00188), the temperature range of the calculation should only be up to 30 degC, which we have probably been abusing forever.
Martin found that the limit of 35 degC was already being used in the equivalent component of NEMO, so we adopted the same in the ESM1.5.
This way the model is stable for these hot climate experiments.